### PR TITLE
[ADD] description field for payment order

### PR DESCRIPTION
**Pull request by GillesTephaneMeyomesse**
This allows users to easily see payment types

### CODE REVIEW

**Comment by alexis-via** [comment]
I don't understand your comment "This allows users to retrieve payment types"...

**Comment by GillesTephaneMeyomesse** [comment]
bad comment sorry...

**Comment by pedrobaeza** [comment]
What is the purpose of this field? You should change version number accordingly

**Comment by GillesTephaneMeyomesse** [comment]
@pedrobaeza. I think it could be good idea to have description field in payment order form

**Comment by pedrobaeza** [comment]
Well, I see it very meaningless, and more if this fields is not used for anything more except "displaying it". As identification I use the payment order number, which is also the reference that is passed to the transferred account moves.

**Comment by adrienpeiffer** [comment]
@pedrobaeza It's simply to allow the user to describe in what the order of payment concerns and therefore to have a list view with this information. For example, this field could contain a description as January salary. Thus, it is easier to identify a payment order based on a description of its content

File: account_payment_order/models/account_payment_order.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ class AccountPaymentOrder(models.Model):
     move_ids = fields.One2many(
         'account.move', 'payment_order_id', string='Journal Entries',
         readonly=True)
+    description = fields.Text()
```
Comment (adrienpeiffer) [inline account_payment_order/models/account_payment_order.py]: @GillesTephaneMeyomesse A char field will be better

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

**Comment by adrienpeiffer** [comment]
@GillesTephaneMeyomesse Could you sqash your last commit ?

**Comment by GillesTephaneMeyomesse** [comment]
Hi @adrienpeiffer last commit is squashed

**Review by adrienpeiffer — APPROVED** [review]
:+1:

**Comment by alexis-via** [comment]
I don't have a strong opinion about this, but it seems it can be useful. It should also be added to search view of payment orders.

**Review by ThomasBinsfeld — APPROVED** [review]

**Comment by rousseldenis** [comment]
@GillesTephaneMeyomesse Can you review @alexis-via last comment and fix it?

Otherwise LGTM

:+1:

**Comment by GillesTephaneMeyomesse** [comment]
@rousseldenis Thanks this is fixed.

**Comment by rousseldenis** [comment]
@GillesTephaneMeyomesse Ok, but that does not fix the 'search'. It does not allow to specifically search on a description 'My Description'

File: account_payment_order/views/account_payment_order.xml [hunk 2]
```diff
@@ -105,6 +107,7 @@
                 <filter name="date_generated_groupby" string="File Generation Date" context="{'group_by': 'date_generated'}"/>
                 <filter name="date_uploaded_groupby" string="File Upload Date" context="{'group_by': 'date_uploaded'}"/>
                 <filter name="state_groupby" string="State" context="{'group_by': 'state'}"/>
+                <filter name="description_groupby" string="Description" context="{'group_by': 'description'}"/>
```
Comment (ThomasBinsfeld) [inline account_payment_order/views/account_payment_order.xml]: @GillesTephaneMeyomesse You only need to add `<field name="description"/>` in the search view

**Review by ThomasBinsfeld — CHANGES_REQUESTED** [review]

**Comment by GillesTephaneMeyomesse** [comment]
too fast, sorry^^

File: account_payment_order/views/account_payment_order.xml (hunk 2 above)
Comment (adrienpeiffer) [reply account_payment_order/views/account_payment_order.xml]: the groupby on the description isn't necessary

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

File: account_payment_order/views/account_payment_order.xml [hunk 3]
```diff
@@ -95,6 +97,7 @@
     <field name="model">account.payment.order</field>
     <field name="arch" type="xml">
         <search string="Search Payment Orders">
+            <field name="description" string="My Description"/>
```
Comment (ThomasBinsfeld) [inline account_payment_order/views/account_payment_order.xml]: @GillesTephaneMeyomesse Remove string key

**Review by ThomasBinsfeld — CHANGES_REQUESTED** [review]

**Comment by GillesTephaneMeyomesse** [comment]
@adrienpeiffer group removed and @ThomasBinsfeld string removed

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by adrienpeiffer — APPROVED** [review]
:+1:

**Review by rami-wafaie — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
Please change module version number

**Comment by GillesTephaneMeyomesse** [comment]
Following the modification in this module the version number has been updated

**Pull request merged**